pkgs/by-name/pr/prisma/package.nix

```
@@ -0,0 +1,88 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  nodejs,
+  pnpm_8,
+  prisma-engines,
+  jq,
+  makeWrapper,
+  callPackage,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "prisma";
+  version = "5.18.0";
+
+  src = fetchFromGitHub {
+    owner = "prisma";
+    repo = "prisma";
+    rev = finalAttrs.version;
+    hash = "sha256-BLD2nKryigXr03BCgGwb3PnCcBLMyDfSFb9Snj0VPKI=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm_8.configHook
+    makeWrapper
+    jq
+  ];
+
+  pnpmDeps = pnpm_8.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-lgdJk7HCfX3cAvdEI8xG/IVBiLWezdUN0q+e/0LtVUQ=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    jq --arg version $version '.version = $version' packages/cli/package.json > packages/cli/package.json.new
+    mv packages/cli/package.json.new packages/cli/package.json
+
+    pnpm build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib/node_modules/prisma}
+    cp -r {node_modules,package.json} $out/lib/node_modules/prisma
+
+    for package in packages/*; do
+      mkdir -p $out/lib/node_modules/prisma/$package
+
+      cp -r $package/{node_modules,package.json} $out/lib/node_modules/prisma/$package
+
+      if [ -d $package/dist ]; then
+        cp -r $package/dist $out/lib/node_modules/prisma/$package
+      elif [ -d $package/build ]; then
+        cp -r $package/build $out/lib/node_modules/prisma/$package
+      fi
+    done
+
+    makeWrapper "${lib.getExe nodejs}" "$out/bin/prisma" \
+      --add-flags "$out/lib/node_modules/prisma/packages/cli/build/index.js" \
+      --set PRISMA_SCHEMA_ENGINE_BINARY ${prisma-engines}/bin/schema-engine \
+      --set PRISMA_QUERY_ENGINE_BINARY ${prisma-engines}/bin/query-engine \
+      --set PRISMA_QUERY_ENGINE_LIBRARY ${lib.getLib prisma-engines}/lib/libquery_engine.node \
+      --set PRISMA_FMT_BINARY ${prisma-engines}/bin/prisma-fmt
```

```suggestion
      --set PRISMA_QUERY_ENGINE_LIBRARY ${lib.getLib prisma-engines}/lib/libquery_engine.node
```

`PRISMA_FMT_BINARY` doesn't exist anymore since quite a while, it has been replaced with a WASM module around two years ago. I'm not sure why `prisma-fmt` binary even exists, maybe because of some third-party clients, Prisma itself is not using it in any way.